src/bench/addrman.cpp

```diff
@@ -70,13 +70,22 @@ static void FillAddrMan(CAddrMan& addrman)
 
 static void AddrManAdd(benchmark::Bench& bench)
 {
-    CreateAddresses();
+     /* Create many CAddrMan objects - one to be modified at each loop iteration. */
 
-    CAddrMan addrman(/* deterministic */ false, /* consistency_check_ratio */ 0);
+    bench.epochs(5).epochIterations(1);
+    size_t addrman_count = bench.epochs() * bench.epochIterations();
+
+    std::vector<std::unique_ptr<CAddrMan>> addrmans(addrman_count);
```

Comment:
not a huge difference here, but could use `reserve` (optional) and `emplace` to avoid unique_ptr?

Reply:
I did try that, but couldn't get it to work. `emplace()` requires the type to be [MoveInsertable](https://en.cppreference.com/w/cpp/named_req/MoveInsertable), which `CAddrMan` isn't since its move constructor is implicitly deleted.

If you have a suggestion for how to do this, I'd be happy to incorporate it.